runelite-client/src/main/java/net/runelite/client/util/DeferredEventBus.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018, Adam <Adam@sigterm.info>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.util;
+
+import com.google.common.eventbus.EventBus;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class DeferredEventBus extends EventBus
+{
+	private final EventBus eventBus;
+	private final Queue<Object> pendingEvents = new ConcurrentLinkedQueue<>();
+
+	@Inject
+	private DeferredEventBus(EventBus eventBus)
+	{
+		this.eventBus = eventBus;
+	}
+
+	@Override
+	public void register(Object object)
+	{
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public void post(Object object)
+	{
+		pendingEvents.add(object);
+	}
+
+	public void post()
+	{
+		int size = pendingEvents.size();
+		while (size-- > 0)
```

Comment:
This will work, supposing nothing else polls this queue.
But this still strikes me as inherently prone to failure.

Reply:
I was trying to avoid the case where an event handler posts something to it, causing it to loop indefinitely.